src/culture/rebels.cpp

```diff
@@ -878,45 +877,36 @@ void rebel_risings_check(sys::state& state) {
 			if(new_to_make == 0)
 				continue;
 
-			ve::execute_serial_fast<dcon::province_id>(state.world.province_size(), [&](auto index) {
-				province_damage.set(index, ve::fp_vector{});
-			});
-
 			/*
 			- When a rising happens, pops with at least define:MILITANCY_TO_JOIN_RISING will spawn faction-organization x
 			max-possible-supported-regiments, to a minimum of 1 (if any more regiments are possible).
 			*/
-
-			
-
-			float total_damage = 0.0f;
 			for(auto pop : rf.get_pop_rebellion_membership()) {
 				if(counter == 0)
 					break;
 
 				if(pop.get_pop().get_militancy() >= state.defines.mil_to_join_rising) {
 					auto location = pop.get_pop().get_province_from_pop_location();
-
 					// this is the logic we would use if we were creating rebel regiments
 					auto max_count = int32_t(state.world.pop_get_size(pop.get_pop()) / (province::is_overseas(state, pop.get_pop().get_province_from_pop_location()) ? (state.defines.pop_min_size_for_regiment_colony_multiplier * state.defines.pop_size_per_regiment) : state.defines.pop_size_per_regiment));
 					auto cregs = pop.get_pop().get_regiment_source();
 					auto used_count = int32_t(cregs.end() - cregs.begin());
-
 					if(used_count < max_count) {
 						auto pop_location = pop.get_pop().get_province_from_pop_location();
-
-						auto new_reg = military::create_new_regiment(state, dcon::nation_id{}, state.military_definitions.irregular);
 						auto a = [&]() {
 							for(auto ar : state.world.province_get_army_location(pop_location)) {
-								if(!(ar.get_army().get_battle_from_army_battle_participation()) && ar.get_army().get_controller_from_army_rebel_control() == rf)
+								if(ar.get_army().get_controller_from_army_rebel_control() == rf && !ar.get_army().get_black_flag() && !ar.get_army().get_is_retreating() && !ar.get_army().get_battle_from_army_battle_participation())
```

Comment:
rebels armies can't be black flagged. And it is fine to insert a new regiment into a retreating army (cheaper than creating a new army, after all)